crates/oxc_isolated_declarations/src/class.rs

```diff
@@ -404,6 +406,7 @@ impl<'a> IsolatedDeclarations<'a> {
                     }
 
                     if method.accessibility.is_some_and(TSAccessibility::is_private) {
+                        self.errors.borrow_mut().truncate(errors_length);
```

Comment:
This is a little bit hacky. I think a less hacky way might be to decouple methods like `transform_formal_parameters` from generating errors and then move the check for errors after this block, but I wanted to get some feedback before starting to attempt something like that.

Reply:
I think we can handle the private method early, can you try it?